entrypoint.sh

```diff
@@ -41,5 +45,22 @@ if ! nexus-allowlist --admin-password "$NEXUS_ADMIN_PASSWORD" --nexus-host "$NEX
     exit 1
 fi
 
-# Run allowlist configuration now, and again whenever allowlist files are modified
-find "$ALLOWLIST_DIR"/*.allowlist | entr -n nexus-allowlist --admin-password "$NEXUS_ADMIN_PASSWORD" --nexus-host "$NEXUS_HOST" --nexus-port "$NEXUS_PORT" update-allowlists --packages "$NEXUS_PACKAGES" --pypi-package-file "$PYPI_ALLOWLIST" --cran-package-file "$CRAN_ALLOWLIST"
+if [ -n "$ENTR_FALLBACK" ]; then
+    echo "$(timestamp) Using fallback file monitoring"
+    # Run allowlist configuration now
+    nexus-allowlist --admin-password "$NEXUS_ADMIN_PASSWORD" --nexus-host "$NEXUS_HOST" --nexus-port "$NEXUS_PORT" update-allowlists --packages "$NEXUS_PACKAGES" --pypi-package-file "$PYPI_ALLOWLIST" --cran-package-file "$CRAN_ALLOWLIST"
+    # Periodically check for modification of allowlist files and run configuration again when they are
+    hash=$(hashes)
+    while true; do
+        new_hash=$(hashes)
+        if [ "$hash" != "$new_hash" ]; then
+            nexus-allowlist --admin-password "$NEXUS_ADMIN_PASSWORD" --nexus-host "$NEXUS_HOST" --nexus-port "$NEXUS_PORT" update-allowlists --packages "$NEXUS_PACKAGES" --pypi-package-file "$PYPI_ALLOWLIST" --cran-package-file "$CRAN_ALLOWLIST"
```

Comment:
We should wrap this stuff in a function rather than replicating it.

Reply:
Yes, I don't like it.

It seems that the argument for `entr` needs to be a file though. So, at least for the `entr` command, you cannot substitute the command or use a function defined in the script.

I'm not sure whether partially duplicating the command is much better than always duplicating it 🤔.

Reply:
Could we not use a function like this?

```bash
update_allowlists() {
    nexus-allowlist --admin-password "$NEXUS_ADMIN_PASSWORD" --nexus-host "$NEXUS_HOST" --nexus-port "$NEXUS_PORT" update-allowlists --packages "$NEXUS_PACKAGES" --pypi-package-file "$PYPI_ALLOWLIST" --cran-package-file "$CRAN_ALLOWLIST"
}
```

Reply:
[It seems not](https://github.com/eradman/entr/issues/6).

I'm not sure moving the command to a new script is much better as you would need to pass some environment variables as arguments, which may make what is going on less clear.

Reply:
Fair enough. Let's deal with the duplication then.